### ISSUE TITLE
Feature Request: Add index.html for New Folders

### ISSUE THREAD

**Issue by salahjaafar** [prasathmani/tinyfilemanager#1192]
Dear TinyFileManager Team,

I'm a big fan of TinyFileManager and use it regularly. I'd like to suggest a small but helpful feature: adding an index.html file automatically when creating a new folder. This would prevent directory listing and improve security for those who expose TinyFileManager to the web.

Thank you for considering my suggestion.

Sincerely,
Salah

**Comment by salahjaafar** [comment]
I've developed a small addition to address this.

Here's the code to add in the tinyfilemanager.php file:

```
// after the line elseif (@is_dir($new_path) && $file != '.' && $file != '..' && fm_is_exclude_items($file)) {
$indexFile = $new_path . '/index.html';
if (!file_exists($indexFile)) {
    $newFile = fopen($indexFile, 'w');
    fwrite($newFile, '<!DOCTYPE html><html><head><title>Index</title></head><body></body></html>');
    fclose($newFile);
}
```


This code creates a basic index.html file in any directory that doesn't have one, which can be helpful to prevent directory listing vulnerabilities and improve navigation.

I hope this contribution is useful to you.

Best regards,

Salah

**Comment by ZeroSuf3r** [comment]
Did something similar, but in my case i'm adding **index.php**.

```
                $fp_index = fopen($path . '/' . $new . '/' . "index.php", 'w');
                fwrite($fp_index, "<?php require_once('" . $root_path . "/index.php'); exit();?>");
                fclose($fp_index);
```

**Comment by salahjaafar** [comment]
> Did something similar, but in my case i'm adding **index.php**.
> 
> ```
>                 $fp_index = fopen($path . '/' . $new . '/' . "index.php", 'w');
>                 fwrite($fp_index, "<?php require_once('" . $root_path . "/index.php'); exit();?>");
>                 fclose($fp_index);
> ```

Yes, the choice is yours, but index.html files are simple static files, whereas PHP requires additional processing by the Apache server.

**Comment by salahjaafar** [comment]
@prasathmani 
I have improved the code to check if there is also an index.php. Do not create an index.html file because it causes me problems with applications that run in the folder and sub folders.
```
// after the line elseif (@is_dir($new_path) && $file != '.' && $file != '..' && fm_is_exclude_items($file)) {
$indexHtmlFile = $new_path . '/index.html';
$indexPhpFile = $new_path . '/index.php';

if (!file_exists($indexHtmlFile) && !file_exists($indexPhpFile)) { // Check both files
    $newFile = fopen($indexHtmlFile, 'w');
    fwrite($newFile, '<!DOCTYPE html><html><head><title>Index</title></head><body></body></html>');
    fclose($newFile);
}
```
